pkg/api/norman/customization/cluster/validator.go

```diff
@@ -485,6 +490,29 @@ func validateAKSClusterName(client v3.ClusterInterface, spec *v32.ClusterSpec) e
 	return nil
 }
 
+// validateAKSNetworkPolicy performs validation around setting enableNetworkPolicy on AKS clusters which turns on Project Network Isolation
+func (v *Validator) validateAKSNetworkPolicy(clusterSpec *v32.ClusterSpec, prevCluster *v3.Cluster) error {
+	// determine if network policy is enabled on the AKS cluster by checking the cluster spec and then the upstream spec if the field is nil (unmanaged)
+	var netPolEnabled bool
+	if clusterSpec.AKSConfig != nil && clusterSpec.AKSConfig.NetworkPolicy != nil {
+		netPolEnabled = to.String(clusterSpec.AKSConfig.NetworkPolicy) == string(containerservice.NetworkPolicyAzure) || to.String(clusterSpec.AKSConfig.NetworkPolicy) == string(containerservice.NetworkPolicyCalico)
+	} else if prevCluster != nil && prevCluster.Status.AKSStatus.UpstreamSpec != nil && prevCluster.Status.AKSStatus.UpstreamSpec.NetworkPolicy != nil {
+		netPolEnabled = to.String(prevCluster.Status.AKSStatus.UpstreamSpec.NetworkPolicy) == string(containerservice.NetworkPolicyAzure) || to.String(prevCluster.Status.AKSStatus.UpstreamSpec.NetworkPolicy) == string(containerservice.NetworkPolicyCalico)
+	} else {
+		return nil
+	}
+
+	// network policy enabled on the AKS cluster is a prerequisite for PNI
+	if enableNetPol := clusterSpec.EnableNetworkPolicy; enableNetPol != nil && *enableNetPol && !netPolEnabled {
```

Comment:
```suggestion
	if to.Bool(clusterSpec.EnableNetworkPolicy) && networkPolicy != string(containerservice.NetworkPolicyAzure) && networkPolicy != string(containerservice.NetworkPolicyCalico) {
```

Reply:
Why not use `to.Bool` here?

Reply:
Arg when I rebased against release/v2.6 I accidentally reverted to the old version